lib/setopt.c

```diff
@@ -863,14 +863,18 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option, va_list param)
       ;
     else
 #endif
-#ifndef USE_NGHTTP2
+#if !defined(USE_NGHTTP2) && !defined(USE_HYPER)
     if(arg >= CURL_HTTP_VERSION_2)
       return CURLE_UNSUPPORTED_PROTOCOL;
 #else
     if(arg >= CURL_HTTP_VERSION_LAST)
       return CURLE_UNSUPPORTED_PROTOCOL;
     if(arg == CURL_HTTP_VERSION_NONE)
       arg = CURL_HTTP_VERSION_2TLS;
+#endif
+#ifdef USE_HYPER
+    if(arg == CURL_HTTP_VERSION_1_0)
```

Comment:
I've just added `hyper_request_set_version(req, HYPER_HTTP_VERSION_1_0)`, which will enforce 1.0 semantics on a 1.x connection.

Reply:
Lovely, I'm adapting and will backpedal this commit. Thanks!